### ISSUE TITLE
Config file in $HOME is not identified

### ISSUE THREAD

**Issue by rogerluan** [igorshubovych/markdownlint-cli#317]
Hi 👋 

I'm trying this out and I placed my `.markdownlintrc` in `$HOME/.markdownlintrc` (aka `~/.markdownlintrc`), but it's not able to pick it up. I read the config placement rules in https://www.npmjs.com/package/rc#standards but that seems to be related to npm projects, which have a name, so I think it doesn't really apply. My `$HOME` is a parent folder of the current directory, and there are no other `.markdownlint*` files anywhere in my computer.

Any hints of what could be the issue?

Thanks in advance!

**Comment by DavidAnson** [comment]
That should work. How do you know it's not being used? Can you show an example?

**Comment by rogerluan** [comment]
So I have a README file that has line length (`MD013`) issues. If I run

```sh
markdownlint -- README.md
```

Then it points out those `MD013` issues.

But if I run:

```sh
markdownlint --config $HOME/.markdownlintrc -- README.md
```

Then it correctly picks up my config where I've set `MD013: false`, then it doesn't throw linter errors for line length.

My config file is identical to the sample/template one in the documentation, except I replaced the MD013 config with:

```yaml
# MD013/line-length - Line length
MD013: false
```

**Comment by rogerluan** [comment]
For documentation purposes, the directly where I'm running the command is

```sh
$HOME/Documents/Projects/some-folder/
```

And the `.markdownlintrc` file is in `$HOME`

**Comment by DavidAnson** [comment]
That all sounds right to me, tho I don't do this myself. I will give it a try later today. What platform, by the way?

**Comment by rogerluan** [comment]
macOS 12.4, do you need any other env dump? I installed via brew btw. No node.js project

**Comment by DavidAnson** [comment]
The [INI format](https://en.wikipedia.org/wiki/INI_file) uses `=`, not `:`, so your `.markdownlintrc` should be:

```ini
# MD013/line-length - Line length
MD013 = false
```

If you make that change, it should behave as you expect.

Interestingly, things work when you pass `--config` because that uses the `markdownlint` library to parse the configuration file which supports `YAML` where `:` is valid.

**Comment by rogerluan** [comment]
@DavidAnson The entire file is in yaml though 😅 You can find it here: https://github.com/rogerluan/dotfiles/blob/master/.markdownlintrc

Just as a side note now that I shared my actual file with you ☝️ my dotfiles repo [uses a symlink](https://github.com/rogerluan/dotfiles/blob/a5ca432f3311dbdfc301d20f20a11d192f43ea6c/set_up_symlinks.sh#L16) for that `.markdownlintrc` file, but that doesn't matter because I also tested this by placing the actual file in that location (without a symlink) and the results were the same 😬

**Comment by rogerluan** [comment]
And just as an extra data point, I pulled the sample from the yaml sample, not ini: https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.yaml

**Comment by rogerluan** [comment]
I also looked for a debugging option like `--verbose`, `--debug`, etc, to hopefully print more logs when executing the CLI, but there's none 😞 is there any env var I can set when executing it, to print logs that would help us debug this issue?

**Comment by rogerluan** [comment]
@DavidAnson I may have misunderstood your previous message though, are you saying that when I place my config file in the $HOME dir, this specific setup only accepts an INI format? 😥

**Comment by DavidAnson** [comment]
Right, the rc/run-con package supports application-named config files in JSON and INI format. It does not seem to be extensible to add other formats. This is unique to CLI(1).

The --config option supports JSON(C) and YAML using a shared function from markdownlint that is common to the library, CLI(1), CLI2, and the VS Code extension. It is extensible.

I'm not a fan of rc/run-con, but it wasn't my doing and it would be breaking to remove.

If you are happy with YAML, you could put (or symlink) a file called .markdownlint.yaml in the current directory when running the tool and it should be picked up automatically.

**Comment by rogerluan** [comment]
Ah, if it's able to pick up json, I'll try that! Just am not super excited about INI 🙈 
Thank you!!

**Comment by rogerluan** [comment]
It worked!! I cannot name it `.json` though, I gotta keep the naming `.markdownlintrc` strictly 😄 but I'm glad we made it work!

I'll add some docs around this just so it doesn't catch anyone else :)
